### PR TITLE
Add app.nestful.Nestful

### PR DESCRIPTION
**Pull request by ofekd**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.

Nestful is a list-taking app that can be used either locally on the user's filesystem or online with a subscription. As the name suggests, it allows the user to nest lists inside other lists, enabling a plethora of use cases that are not as straightforward to achieve in other app, such as "bubbling" due items, allowing to compare priorities between unrelated projects sharing the user's time.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by ofekd** [comment]
Note: I've received these 2 screenshots errors:

```
{
    "errors": [
        "appstream-screenshots-files-not-found-in-ostree",
        "appstream-screenshots-not-mirrored-in-ostree"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

But reading the link I gathered it may be solved automatically by the build bot. Please let me know if corrections are needed on my end


Thanks

**Comment by hfiguiere** [comment]
> Please describe the application briefly.

this is missing while you checked it.

File: app.nestful.Nestful.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - type: file
+        dest-filename: Nestful.deb
+        url: https://cdn.crabnebula.app/download/nestful/nestful/latest/platform/debian-x86_64
+        sha256: aea487c12f2fdf85d22b9fea80f36aa237d3e5897f8301f9d7b7152c5e9e87aa
+        only-arches: [x86_64]
+    build-commands:
+      - ar -x *.deb
+      - tar -xf data.tar.gz
+      - 'install -Dm755 usr/bin/nestful /app/bin/nestful'
+      - sed -i 's/Icon=nestful/Icon=app.nestful.Nestful/'  usr/share/applications/Nestful.desktop
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:30]: don't use sed. the command `desktop-file-edit` exist for that.

File: app.nestful.Nestful.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        path: ./app.nestful.Nestful.metainfo.xml
+      - type: file
+        dest-filename: Nestful.deb
+        url: https://cdn.crabnebula.app/download/nestful/nestful/latest/platform/debian-x86_64
+        sha256: aea487c12f2fdf85d22b9fea80f36aa237d3e5897f8301f9d7b7152c5e9e87aa
+        only-arches: [x86_64]
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:25]: so what happen on aarch64?

File: app.nestful.Nestful.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  - --socket=fallback-x11 # Permission needed to show the window
+  - --device=dri # OpenGL, not necessary for all projects
+  - --share=ipc
+
+modules:
+  - name: binary
+    buildsystem: simple
+    sources:
+      - type: file
+        path: ./app.nestful.Nestful.metainfo.xml
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:20]:
```suggestion
        path: app.nestful.Nestful.metainfo.xml
```

File: app.nestful.Nestful.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  - proprietary
+command: nestful
+finish-args:
+  - --socket=wayland # Permission needed to show the window
+  - --socket=fallback-x11 # Permission needed to show the window
+  - --device=dri # OpenGL, not necessary for all projects
+  - --share=ipc
+
+modules:
+  - name: binary
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:16]: use a name that actually represent what it install. like the actual app name.

File: app.nestful.Nestful.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+runtime: org.gnome.Platform
+runtime-version: '47'
+sdk: org.gnome.Sdk
+tags:
+  - proprietary
+command: nestful
+finish-args:
+  - --socket=wayland # Permission needed to show the window
+  - --socket=fallback-x11 # Permission needed to show the window
+  - --device=dri # OpenGL, not necessary for all projects
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:12]:
```suggestion
  - --socket=wayland
  - --socket=fallback-x11
  - --device=dri
```

File: app.nestful.Nestful.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+id: app.nestful.Nestful
+
+runtime: org.gnome.Platform
+runtime-version: '47'
```
Comment (hfiguiere) [inline app.nestful.Nestful.yml:4]: do you really need the GNOME runtime?

File: app.nestful.Nestful.metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  <metadata_license>CC-BY-SA-4.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://nestful.app/terms</project_license>
+  <developer id="app.nestful">
+    <name>Nestful</name>
+  </developer>
+  <url type="homepage">https://nestful.app</url>
+
+  <icon type="stock">app.nestful.Nestful</icon>
+  <icon type="remote">https://nestful.app/img/icons/android-chrome-512x512.png</icon>
+
```
Comment (hfiguiere) [inline app.nestful.Nestful.metainfo.xml:17]:
```suggestion
```

File: app.nestful.Nestful.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>app.nestful.Nestful</id>
+
+  <name>Nestful</name>
+  <summary>The single place for all your lists and tasks</summary>
+
+  <metadata_license>CC-BY-SA-4.0</metadata_license>
+  <project_license>LicenseRef-proprietary=https://nestful.app/terms</project_license>
```
Comment (hfiguiere) [inline app.nestful.Nestful.metainfo.xml:9]: is this redistributable? If it is not it needs to use `extra-data`

File: app.nestful.Nestful.metainfo.xml [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+  </description>
+
+  <launchable type="desktop-id">app.nestful.Nestful.desktop</launchable>
+  <screenshots>
+    <screenshot type="default">
+      <caption>Easily arrange projects</caption>
+      <image>https://nestful.app/img/landing/screenplay-outline.png</image>
+    </screenshot>
+    <screenshot>
+      <caption>Due items bubble up for your attentio</caption>
```
Comment (hfiguiere) [inline app.nestful.Nestful.metainfo.xml:71]:
```suggestion
      <caption>Due items bubble up for your attention</caption>
```

File: app.nestful.Nestful.yml (hunk 6 above)
Comment (ofekd) [reply app.nestful.Nestful.yml:4]: Yes, and specifically 47 -- earlier versions did not function correctly

File: app.nestful.Nestful.yml (hunk 2 above)
Comment (hfiguiere) [reply app.nestful.Nestful.yml:25]: but you still build an aarch64 package.

File: app.nestful.Nestful.metainfo.xml (hunk 8 above)
Comment (ofekd) [reply app.nestful.Nestful.metainfo.xml:9]: I've converted to `extra-data` but `ar` seems to not be available -- what is the preferred way to have it be available to `apply_extra`?

Comment (bbhtt) [reply app.nestful.Nestful.metainfo.xml:9]: https://github.com/flathub/com.amazon.Workspaces/blob/2f1b8c8bb75c1bf3f24fd0896a3e5844de985fe5/com.amazon.Workspaces.yml#L140-L142

File: app.nestful.Nestful.yml (hunk 2 above)
Comment (bbhtt) [reply app.nestful.Nestful.yml:25]: https://docs.flathub.org/docs/for-app-authors/requirements#flathubjson

File: app.nestful.Nestful.metainfo.xml (hunk 8 above)
Comment (bbhtt) [reply app.nestful.Nestful.metainfo.xml:9]: You have to pull the desktop file and icon as well here, pick one icon. See https://docs.flatpak.org/en/latest/module-sources.html#extra-data

Comment (ofekd) [reply app.nestful.Nestful.metainfo.xml:9]: Thanks, I've implemented that. By the time I saw this comment I already added the different-sized icons (I don't have a vector version) -- let me know if you want me to leave only the largest one.

File: app.nestful.Nestful.desktop [hunk 10]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Categories=Office;
+Comment=The Nestful app
+Exec=/app/bin/nestful
```
Comment (bbhtt) [inline app.nestful.Nestful.desktop:4]:
```suggestion
Exec=nestful
```

File: app.nestful.Nestful.yml [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+  - proprietary
+command: nestful
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --device=dri
+  - --share=ipc
+
+modules:
+  - name: binutils
```
Comment (bbhtt) [inline app.nestful.Nestful.yml:16]:
```suggestion
  - name: nestful
```

Comment (ofekd) [reply app.nestful.Nestful.yml]: Sorry, leftover experiment -- removed

**Comment by ofekd** [comment]
Thanks @bbhtt 

I've tested and network access is missing. Is there a way to ask for permission programmatically, only when the user attempts to use network features?

I don't want to add this by default for users who are offline-only, but will turn on by default if it cannot be requested dynamically

**Comment by bbhtt** [comment]
No you have to add `--share=network` there is no other dynamic thing

**Comment by ofekd** [comment]
@bbhtt Added, thank you. If there's an existing discussion/issue regarding dynamic permissions I'd love a link to read

**Comment by bbhtt** [comment]
bot, build app.nestful.Nestful

**Comment by ofekd** [comment]
@bbhtt Looks good, thank you!

**Pull request closed without merging**